salt/orch/etcd-migrate.sls

```diff
@@ -0,0 +1,107 @@
+# Generic Updates
+update_pillar:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_pillar
+
+update_grains:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_grains
+
+update_mine:
+  salt.function:
+    - tgt: '*'
+    - name: mine.update
+    - require:
+       - salt: update_pillar
+       - salt: update_grains
+
+{%- set masters = salt.saltutil.runner('mine.get', tgt='G@roles:kube-master', fun='network.ip_addrs', tgt_type='compound') %}
+{%- for master_id in masters.keys() %}
+
+{{ master_id }}-stop-services:
+  salt.state:
+    - tgt: {{ master_id }}
+    - sls:
+      - kube-apiserver.stop
+      - kube-controller-manager.stop
+      - kube-scheduler.stop
+      - etcd.stop
+
+{% endfor %}
+
+{%- set workers = salt.saltutil.runner('mine.get', tgt='G@roles:kube-minion', fun='network.ip_addrs', tgt_type='compound') %}
+{%- for worker_id, ip in workers.items() %}
+
+{{ worker_id }}-stop-services:
+  salt.state:
+    - tgt: {{ worker_id }}
+    - sls:
+      - kubelet.stop
+      - kube-proxy.stop
```

Comment:
kube proxy also exists on the masters. However, do we need to stop it? Stopping it means all the traffic into apps is stopped.

Reply:
All Master nodes are removed from scheduling. I can stop also kube-proxy on Master, but it doesn't have any influence

Reply:
Scheduling status has no effect on kube-proxy, if we need to stop it, we need to stop it on the masters too. However, I don't believe we do need to stop it, and stopping it causes further unnecessary disruption to the cluster during the migration (specifically - it's preventing traffic from reaching any of the pods which are already running in the cluster)